cookbooks/kinematically_driven_subduction_2d/doc/kinematically_driven_subduction_2d.md

```diff
@@ -100,3 +99,98 @@ boundary. This can also be seen in {numref}`fig:QQ_case1_results`.
+two plugins called "composition_trms_statistics" and "isotherm depth".
+To this end, we add the postprocessors "composition RMS temperature statistics"
+and "isotherm depth" to the prm file. The RMS temperature is tracked
+over the subducting plate (comprising BOC_SP, SHB_SP, thermal_SP)
+and the whole domain over time. And the isotherm temperature to be
+tracked is prescribed as 800 &degC.
+
+```{literalinclude} Case2a_postprocessing.prm
+
+```
+
+{numref}`fig:QQ_case2a_diagnostics` shows the RMS temperature,
+800 &degC isotherm depth as well as the other diagnostics. The 800 &degC isotherm
+depth continuously increases over time, but towards the end at a slower rate.
+The RMS temperature of the whole domain keeps decreasing due to the
+input of cold subducting plate into the model domain. The slab RMS temperature
+decreases at the initial stage, then it goes up due to the slab being warmed up by the
+mantle. Note that the other diagnostics are indeed the same as for Case 1.
+
+```{figure-md} fig:QQ_case2a_diagnostics
+<img src="Case2a_diagnostics.*" />
+
+ ASPECT Case 2a and Case 1, Elefant Case 2a diagnostic quantities
+```
+
+## Case 2b: Model with a temperature-dependent density
+
+Case 2b builds on Case 2a.  Compared to Case 2a, the thermal expansivity in Case 2b
+is set to 2.5e-5 K<sup>-1</sup> instead of 0 K<sup>-1</sup>. Therefore, temperature feeds into the density
+and therefore into the Stokes equations. In addition, the benchmark uses different reference temperatures for each composition, but only one value can be specified for the reference temperature
+in ASPECT. Therefore, we adapted the reference densities for the different compositional fields to reflect the different reference temperatures. All the other parameters are
+identical to Case 2a:
+
+```{literalinclude} Case2b_materialmodel.prm
+
+```
+The density evolution can be seen in {numref}`fig:QQ_case2b_density_evolution`. When the slab warms up, the density contrast between the slab and the mantle decreases.
+
+```{figure-md} fig:QQ_case2b_density_evolution
+<img src="Case2b_density_evolution.*" />
+
+ Case2b density evolution
+```
+
+Therefore, the  subduction in this case is much slower than that in Case 1 and 2a (see {numref}`fig:QQ_case2b_diagnostics`). The subducting plate does not reach
+the bottom in 15 Myr, i.e., the third stage of Case 1 and 2a is missing in this
+case.
+
+```{figure-md} fig:QQ_case2b_diagnostics
+<img src="Case2b_diagnostics.*" />
+
+ ASPECT Case2b diagnostic quantities
+```
+
+The RMS velocity over the whole domain is overall higher than the previous cases (see {numref}`fig:QQ_case2b_diagnostics`). This is due to higher sub-lithospheric velocities on the
+left side of the domain where a second convection cell forms (see {numref}`fig:QQ_case2b_velocity_evolution`)
+
+```{figure-md} fig:QQ_case2b_velocity_evolution
+<img src="Case2b_velocity_evolution.*" />
+
+ Case2b velocity evolution
+```
```

Comment:
may be add here a cliffhanger? i.e. something like, of course case 2b remains a very simplified model as all compositions are linear viscous. We will address this in the Case 3 models which will be added here in the future.